File: AVLTrees/examples/avl_tree_example.js
const AVLTree = require('../source/avl_tree'); // Adjust the path based on your project structure

// Initialize the AVL Tree
const tree = new AVLTree();

// Insert values into the AVL Tree
const valuesToInsert = [10, 20, 5, 15, 30, 25, 35, 16];
console.log('Inserting values:', valuesToInsert);
valuesToInsert.forEach(value => tree.insert(value));

// Print the tree in in-order traversal (should be sorted)
console.log('In-order traversal after insertions:');
tree.printInOrder();

// Search for a value in the tree
const searchValue = 15;
console.log(`Searching for value ${searchValue} in the tree...`);
if (tree.search(searchValue)) {
    console.log(`Value ${searchValue} found in the tree.`);
} else {
    console.log(`Value ${searchValue} not found in the tree.`);
}

// Delete a value from the AVL Tree
const deleteValue = 20;
console.log(`Deleting value ${deleteValue} from the tree...`);
tree.delete(deleteValue);

// Print the tree in in-order traversal after deletion
console.log('In-order traversal after deletion:');
tree.printInOrder();

// Insert additional values to test rebalancing
const additionalValues = [18, 3, 40];
console.log('Inserting additional values:', additionalValues);
additionalValues.forEach(value => tree.insert(value));

// Print the tree in in-order traversal after additional insertions
console.log('In-order traversal after additional insertions:');
tree.printInOrder();

File: AVLTrees/examples/avl_tree_example.cpp
#include "../source/avl_tree.h"
#include <iostream>

int main() {
    AVLTree tree;

    // Insert values into the AVL tree
    std::cout << "Inserting values into the AVL Tree: 10, 20, 5, 15, 30, 25, 35, 16\n";
    tree.insert(10);
    tree.insert(20);
    tree.insert(5);
    tree.insert(15);
    tree.insert(30);
    tree.insert(25);
    tree.insert(35);
    tree.insert(16);

    // Print the tree in-order
    std::cout << "In-order traversal after insertions:\n";
    tree.printInOrder();

    // Search for a value in the tree
    int searchValue = 15;
    std::cout << "Searching for value " << searchValue << " in the AVL Tree...\n";
    if (tree.search(searchValue)) {
        std::cout << "Value " << searchValue << " found in the AVL Tree.\n";
    } else {
        std::cout << "Value " << searchValue << " not found in the AVL Tree.\n";
    }

    // Delete a value from the AVL tree
    int deleteValue = 20;
    std::cout << "Deleting value " << deleteValue << " from the AVL Tree...\n";
    tree.remove(deleteValue);

    // Print the tree in-order after deletion
    std::cout << "In-order traversal after deletion:\n";
    tree.printInOrder();

    // Insert additional values to test rebalancing
    std::cout << "Inserting additional values into the AVL Tree: 18, 3, 40\n";
    tree.insert(18);
    tree.insert(3);
    tree.insert(40);

    // Print the tree in-order after additional insertions
    std::cout << "In-order traversal after additional insertions:\n";
    tree.printInOrder();

    return 0;
}